aimmo-game/simulation/location.py

```diff
@@ -1,10 +1,26 @@
 class Location(object):
     def __init__(self, x, y):
-        self.x = x
-        self.y = y
+        self._x = x
+        self._y = y
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def y(self):
+        return self._y
 
     def __add__(self, direction):
-        return Location(self.x + direction.x, self.y + direction.y)
+        try:
+            d_x = direction.x
+            d_y = direction.y
+        except AttributeError:
+            if direction is None:
+                return self
+            raise TypeError('Not a direction.')
```

Comment:
Should return `NotImplemented` in case the other object implements `__radd__`.

Also, should implement `__radd__` (so that `direction + location` works).